### DISCUSSION TITLE
How to install OS from PXE boot?

### DISCUSSION THREAD

**Discussion by metzdemant** [utmapp/UTM#3808] [General]
Hi,
Previously I was using Parallels but colleague told me about UTM so I wanted to try it.
I wanted to install Windows 10 from PXE boot like I was doing previously, but I can't seem to find an option to do that in UTM.
Is that possible? And if yes, how do I do it?

**Comment by danielwjohnston** [comment]
+1 Also trying to pxe boot install for UTM

**Comment by belthesar** [comment]
Heya! This is clearly not a hugely in demand feature, but I agree that this would be a nice to have feature. I was looking to play around with a new OS, and I generally install via netboot.xyz if I can these days, which is super handy for not having to have a library of ISOs on hand). Being able to set up a new VM with PXE booting capability would be quite handy. 

I think I've seen other projects use iPXE to support this functionality, so I wonder if adopting that would be a low-enough lift to make this work?

**Comment by HerbieJay** [comment]
I am also interested in having the ability perform PXE installs. I need to be able to test software in lots of different environment/hardware configurations and find it a lot simpler to have all of the images live on a remote server saving space on my Mac.

**Comment by belthesar** [comment]
I just came back to playing with this after @HerbieJay's thread, and I was able to do a TFTP boot with the UEFI firmware that comes long for the ride with UTM/qemu. In my case, I was using netboot.xyz. Here's what I had to do: 

1. Create a new VM in UTM. I chose the Virtualize option, pick the one that's applicable for your use case. 
2. For Operating System, choose Other. 
3. Choose "None" for your boot device. 
4. Set your CPU, RAM, and Storage values accordingly. On the finalize screen, check "Open VM settings", so we can configure the network settings. 
5. In the VM settings, go to Devices -> Network. Select *Bridged* for the network mode. Choose the appropriate network interface that has access to your DHCP and BOOTP configuration. 
6. Set up the requisite settings in your DHCP, TFTP server, etc. to support BOOTP / TFTP boot
> [!IMPORTANT]
> BOOTP can only point to a single file name to be pulled via TFTP, which means you must provide an EFI executable that is of an architecture supported by your environment. Don't be like me and go "I know netboot.xyz works for PXE booting, why is it not starting?" when you've been booting x86_64 systems all the time, and now you're trying to start an aarch64 VM 🤣
